### PR TITLE
my proposal of functional lib

### PR DESCRIPTION
**Pull request by jhonmike**
- migrating files to functional programming
- fix specs files to import the functional lib
- updated documentation fro use of functional code

issue: #68

### CODE REVIEW

File: src/helpers/luhn.js [hunk 1]
```diff
@@ -1,34 +1,28 @@
...
-    let digit = 0;
-    let i = number.length;
-    let even = true;
+const computed = [0, 2, 4, 6, 8, 1, 3, 5, 7, 9];
 
-    while (i--) {
-      digit = Number(number[i]);
-      sum += (even = !even) ? this.computed[digit] : digit;
-    }
+function _sum(number) {
```
Comment (fernahh) [inline src/helpers/luhn.js]: Remove the `_` here :sweat_smile:

**Comment by fernahh** [comment]
@carlosrberto can you make a code review in this pull request? :smile:

File: src/creditCard.js [hunk 2]
```diff
@@ -1,64 +1,45 @@
...
-import Luhn from './helpers/luhn';
+import CREDIT_CARD_LIST from './creditCardList';
+export { isValid } from './helpers/luhn';
 
-class CreditCard {
-  constructor() {
-    this.creditcardlist = new CreditCardList();
-    this.luhn = new Luhn();
-  }
+export function getCreditCardNameByNumber(number) {
```
[the remark this answers is not in the document]
Comment (jhonmike) [reply src/creditCard.js]: I believe the purpose of this PR is to just pass the lib to a format close to functional, these settings deserve a specific PR

**Comment by fernahh** [comment]
@carlosrberto Can you open another issue with this suggestions about these functional tips?

**Comment by carlosrberto** [comment]
I'll open @fernahh!

**Pull request merged**